src/address.rs

```diff
@@ -1,116 +1,81 @@
 use crate::error::Error;
 use crate::SubnetID;
+use fil_actors_runtime::cbor;
+use fvm_ipld_encoding::RawBytes;
 use fvm_shared::address::Address;
 use fvm_shared::ActorID;
 use lazy_static::lazy_static;
 use serde::{Deserialize, Serialize};
-use std::fmt;
-use std::fmt::Formatter;
 use std::str::FromStr;
 
-/// Maximum length for an address payload determined by
-/// the maximum size of the IPC address.
-const MAX_ADDRESS_LEN: usize = 54;
-
 // The default actor id namespace for IPC addresses
 lazy_static! {
     pub static ref DEFAULT_IPC_ACTOR_NAMESPACE_ID: ActorID = 1000u64;
 }
 
 #[derive(Clone, PartialEq, Eq, Debug, Hash, Serialize, Deserialize)]
 pub struct IPCAddress {
-    inner: Address,
-}
-
-impl fmt::Display for IPCAddress {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        self.inner.fmt(f)
-    }
+    subnet_id: SubnetID,
+    raw_address: Address,
 }
 
 impl IPCAddress {
     /// Generates new address using ID protocol
-    pub const fn new_id(id: u64) -> Self {
+    pub fn new_id(id: u64) -> Self {
+        let d = SubnetID::default();
```

Comment:
To use a relevant name for the variable?
```suggestion
        let sn = SubnetID::default();
```